### PR TITLE
PoS Block Producer updates

### PR DESCRIPTION
**Pull request by AeonSw4n**

### CODE REVIEW

File: lib/pos_blockchain_test.go [hunk 1]
```diff
@@ -1891,8 +1870,6 @@ func testProcessBlockPoS(t *testing.T, testMeta *TestMeta) {
 		var malformedOrphanBlock *MsgDeSoBlock
 		malformedOrphanBlock = _generateRealBlock(testMeta, 18, 18, 9273, testMeta.chain.BlockTip().Hash, false)
 		malformedOrphanBlock.Header.PrevBlockHash = randomHash
-		// Modify anything to make the block malformed, but make sure a hash can still be generated.
-		malformedOrphanBlock.Header.TxnConnectStatusByIndexHash = randomHash
```
Comment (lazynina) [inline lib/pos_blockchain_test.go:1895]: We still need to modify something to make sure the block is malformed such that the test passes.

File: lib/pos_blockchain_test.go [hunk 2]
```diff
@@ -1909,7 +1886,6 @@ func testProcessBlockPoS(t *testing.T, testMeta *TestMeta) {
 		require.True(t, malformedOrphanBlockInIndex.IsStored())
 
 		// If a block can't be hashed, we expect to get an error.
-		malformedOrphanBlock.Header.TxnConnectStatusByIndexHash = nil
```
Comment (lazynina) [inline lib/pos_blockchain_test.go:1912]: Modify something else such that we can't hash the block please.

File: lib/state_change_syncer.go [hunk 3]
```diff
@@ -750,23 +750,6 @@ func (stateChangeSyncer *StateChangeSyncer) SyncMempoolToStateSyncer(server *Ser
 			// If the transaction successfully connected, we update mempoolTxUtxoView to the copied view.
 			if err == nil {
 				mempoolTxUtxoView = copiedView
-			} else {
```
Comment (lazynina) [inline lib/state_change_syncer.go:753]: If err is not nil, we should glog an error maybe?

**Review by lazynina — CHANGES_REQUESTED** [review]

File: lib/state_change_syncer.go (hunk 3 above)
Comment (AeonSw4n) [reply lib/state_change_syncer.go:753]: Agreed. However, I think we should only glog here, similarly to the block producer, which skips over failing transactions now. But I'm not fully familiar with the use of this function, so I may be wrong here.

**Pull request merged**